README.md

```diff
@@ -1,4 +1,18 @@
 FileIO
 ======
 
-A generic C++ file IO class
+Is a collection of file access and directory traversal utilities that was written to be a fast, easy to use and non-exception throwing library. 
+
+The major areas it deals with are:
+
+*_Directory  traversal and file access actions_*:  
+The _DirectoryReader_ and _FileSystemWalker_ are efficient for traversing directory structures and accessing files with ad-hoc actions on those files.  The code was written to be efficient, fast and easy to use. 
+
+Errors are handled through return types and embedded error messages with the result.   In an error-prone scenario the FileIO execution is significantly faster than than solutions that utilizes exceptions. 
```

Comment:
@john-gress, yes, but I didn't call it out specifically.  I will add a link to the Result object.